plotly/plotly/plotly.py

```diff
@@ -1271,6 +1334,50 @@ def validate_credentials(credentials):
         raise exceptions.PlotlyLocalCredentialsError()
 
 
+def add_share_key_to_url(response, **payload):
+    """
+    Update plot's url to include the secret key
+
+    payload keyword agruments:
+    platform ('python') -- the platform that is associated with this url
+    version (string) -- the Plotly version that was used to create the plot
+    args (dict) -- contains the data from figure
+    un (string) -- username from credentials file
+    key (string) -- api_key from credentials file
+    origin('plot') --
+    kwargs (dict) --
+        'filename': (string) the name that is associated with this url
+        'fileopt': ('new' | 'overwrite' | 'extend' | 'append')
+                   file option of the url
+        'world_readable': (bool) make the plot public/ private
+        'sharing': ('public' | 'private' | 'secret')
+                    privacy option of the url
+        'layout': (dict) the layout of the figure
+
+    """
+    domain = (get_session_config().get('plotly_domain') or
+              get_config()['plotly_domain'])
+    plot_id = response['url'].replace(domain + "/~"
+                                      + payload['un'] + "/", "")
+    url = "https://api.plot.ly/v2/files/" + payload['un'] + ":" + plot_id
+    plot_kwargs = json.loads(payload['kwargs'])
+    new_response = requests.patch(url,
+                                  headers={'Plotly-Client-Platform':
+                                           payload['platform']},
+                                  auth=HTTPBasicAuth(payload['un'],
+                                                     payload['key']),
+                                  data={"share_key_enabled":
+                                        "True",
+                                        "world_readable":
+                                        plot_kwargs['world_readable'],
+                                        "filename": plot_kwargs['filename'],
+                                        "fileopt": plot_kwargs['fileopt']})
```

Comment:
need error checking here in case the response fails. `new_response.raise_for_status()`. we should also have friendly error messages around unsupported enterprise releases: if the `url` isn't public plotly *and* the response is a 404 then it's likely the user was using this on an old plotly enterprise without the `share_key` updates. instead of responding with a confusing requests error, we should explicitly tell them that we suspect that they are using an old enterprise version